packages/nimble-components/src/rich-text/editor/tests/rich-text-editor.spec.ts

```diff
@@ -1052,6 +1052,395 @@ describe('RichTextEditor', () => {
+                            () => {
+                                pageObject.pasteHTMLToEditor(value.input);
+
+                                expect(pageObject.getEditorTagNames()).toEqual([
+                                    'P',
+                                    'A'
+                                ]);
+                                expect(
+                                    pageObject.getEditorLeafContents()
+                                ).toEqual([value.url]);
+                                expect(
+                                    pageObject.getEditorLastChildAttribute(
+                                        'href'
+                                    )
+                                ).toBe(value.url);
+                            }
+                        );
+                    }
+                );
+            });
+
+            describe('pasting various links between nodes should render as absolute links between nodes ', () => {
+                const differentProtocolLinks = [
+                    {
+                        name: 'Absolute link(https) within paragraph node',
+                        input: '<p>Anchor between <a href="https://nimble.ni.dev">https://nimble.ni.dev</a> text</p>',
+                        url: 'https://nimble.ni.dev',
+                        textContent: 'Anchor between https://nimble.ni.dev text'
+                    },
+                    {
+                        name: 'Hyperlink(http) within paragraph node',
+                        input: '<p>Anchor between <a href="http://nimble.ni.dev">Nimble</a> text</p>',
+                        url: 'http://nimble.ni.dev',
+                        textContent: 'Anchor between http://nimble.ni.dev text'
+                    },
+                    {
+                        name: 'Absolute link(https) as nimble-anchor within paragraph node',
+                        input: '<p>Anchor between <nimble-anchor href="https://nimble.ni.dev"><a href="https://nimble.ni.dev">https://nimble.ni.dev</a></nimble-anchor> text</p>',
+                        url: 'https://nimble.ni.dev',
+                        textContent: 'Anchor between https://nimble.ni.dev text'
+                    },
+                    {
+                        name: 'Hyperlink(http) as nimble-anchor within paragraph node',
+                        input: '<p>Anchor between <nimble-anchor href="http://nimble.ni.dev"><a href="http://nimble.ni.dev">Nimble</a></nimble-anchor> text</p>',
+                        url: 'http://nimble.ni.dev',
+                        textContent: 'Anchor between http://nimble.ni.dev text'
+                    }
+                ] as const;
+
+                parameterizeNamedList(
+                    differentProtocolLinks,
+                    (spec, name, value) => {
+                        spec(
+                            `${name} renders as absolute link(href and text content should be same) in editor`,
```

Comment:
I just want to point out that these tests would pass even if the returned HTML was `<p><a href="...">Anchor between ... text</a></p>`, which is not what you expect. I'm not saying you need to change these tests, but it may be worth being more precise when writing future tests.

Reply:
I have updated the assertion method as `getEditorTextContents` which returns the array of text contents in the editor. So in this case `<p><a href="...">Anchor between ... text</a></p>`, this newly added method in the page object will return an array with a whole content in both the index.